internal/provider/resources/flow.go

```diff
@@ -0,0 +1,347 @@
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/customtypes"
+)
+
+var (
+	_ = resource.ResourceWithConfigure(&FlowResource{})
+	_ = resource.ResourceWithImportState(&FlowResource{})
+)
+
+// FlowResource contains state for the resource.
+type FlowResource struct {
+	client api.PrefectClient
+}
+
+// FlowResourceModel defines the Terraform resource model.
+type FlowResourceModel struct {
+	ID          types.String               `tfsdk:"id"`
+	Created     customtypes.TimestampValue `tfsdk:"created"`
+	Updated     customtypes.TimestampValue `tfsdk:"updated"`
+	WorkspaceID customtypes.UUIDValue      `tfsdk:"workspace_id"`
+	AccountID   customtypes.UUIDValue      `tfsdk:"account_id"`
+
+	Name types.String `tfsdk:"name"`
+	Tags types.List   `tfsdk:"tags"`
+}
+
+// NewFlowResource returns a new FlowResource.
+//
+//nolint:ireturn // required by Terraform API
+func NewFlowResource() resource.Resource {
+	return &FlowResource{}
+}
+
+// Metadata returns the resource type name.
+func (r *FlowResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_flow"
+}
+
+// Configure initializes runtime state for the resource.
+func (r *FlowResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(api.PrefectClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected provider client type",
+			fmt.Sprintf("Expected api.PrefectClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// Schema defines the schema for the resource.
+func (r *FlowResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	defaultEmptyTagList, _ := basetypes.NewListValue(types.StringType, []attr.Value{})
+
+	resp.Schema = schema.Schema{
+		Description: "The resource `flow` represents a Prefect Cloud Flow. " +
+			"Flows are the most central Prefect object. " +
+			"A flow is a container for workflow logic as-code and allows users to configure how their workflows behave. " +
+			"Flows are defined as Python functions, and any Python function is eligible to be a flow.",
+		Version: 0,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				// We cannot use a CustomType due to a conflict with PlanModifiers; see
+				// https://github.com/hashicorp/terraform-plugin-framework/issues/763
+				// https://github.com/hashicorp/terraform-plugin-framework/issues/754
+				Description: "Flow ID (UUID)",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"created": schema.StringAttribute{
+				Computed:    true,
+				CustomType:  customtypes.TimestampType{},
+				Description: "Timestamp of when the resource was created (RFC3339)",
```

Comment:
can we use a a `stringplanmodifier.UseStateForUnknown()` planmodifier for `created` here?

Reply:
Interesting suggestion. Let me take a look at that to see what the experience is.

Reply:
since flows currently don't support updating, I'm not sure this adds anything to the overall experience. Happy either way.

Reply:
ah that's right, they'll be re-created every time. we can disregard